spec/workers/postcodes_collection_worker_spec.rb

```diff
@@ -0,0 +1,24 @@
+RSpec.describe PostcodesCollectionWorker do
+  describe "#perform" do
+    def perform
+      described_class.new.perform
+    end
+
+    context "where there are postcodes to be refreshed" do
+      before do
+        allow(Postcode).to receive(:select).and_return(%w[E18QS E18QL])
```

Comment:
I don't think `select` returns the values - it returns a scaled back instance of `Postcode`. See equivalent example in Whitehall:

```
Document.limit(2).select(:id)
=> [#<Document:0x00007f60fa796b18 id: 4377>, #<Document:0x00007f60fa796a50 id: 4397>]
```

What you're probably looking for is `pluck`:

```
Document.limit(2).pluck(:id)
=> [4377, 4397]
```

Reply:
Ohh I see, thanks. I wasn't aware of this 🤔